frontend/src/app/reports/reports.component.ts

```diff
@@ -46,31 +34,41 @@ export class ReportsComponent implements OnInit {
     ]);
 
 
-    constructor(private reportsService: ReportsService, private configService: ConfigService) {
+    constructor(private reportsService: ReportsService,
+                private configService: ConfigService,
+                private reportsChartService: ReportsChartService) {
         this.config = configService.getConfig();
     }
 
     ngOnInit() {
-        this.progressBarIsHidden = true;
+        this.progressBarIsHidden = false;
         this.emptyJSONReturned = false;
+        this.charts = this.reportsService.getCharts();
+        // defaults
```

Comment:
Not sure how much value the comment provides. Constructors and "onInit" functions commonly set default values.